packages/core/tracing/src/components/trace-viewer/SpanAttribute.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <ConfigCardItem
+    :item="{
+      type: ConfigurationSchemaType.Text,
+      key: keyValue.key,
+      label: keyValue.key,
+      value: formattedValue,
+    }"
+  >
+    <template
+      v-for="attrKey in ENTITY_LINKED_ATTRIBUTE_KEYS"
+      :key="attrKey"
+      #[attrKey]
+    >
+      <EntityLink
+        v-if="entityLink && entityLinkData"
+        allow-copy
+        :entity-link-data="entityLinkData"
+        :external-link="entityLink"
+        new-window
+      />
+    </template>
+  </ConfigCardItem>
+</template>
+
+<script setup lang="tsx">
+import { ConfigCardItem, ConfigurationSchemaType, EntityLink, type EntityLinkData } from '@kong-ui-public/entities-shared'
+import type { IKeyValue } from '@opentelemetry/otlp-transformer'
+import { computed, inject, onWatcherCleanup, ref, shallowRef, watchEffect, type PropType } from 'vue'
+import { SPAN_ATTRIBUTE_VALUE_UNKNOWN, SPAN_ATTRIBUTES, TRACE_VIEWER_CONFIG } from '../../constants'
+import type { Span, TraceViewerConfig } from '../../types'
+import { getPhaseAndPlugin, unwrapAnyValue } from '../../utils'
+
+import '@kong-ui-public/entities-shared/dist/style.css'
+
+const props = defineProps({
+  span: {
+    type: Object as PropType<Span>,
+    required: true,
+  },
+  keyValue: {
+    type: Object as PropType<IKeyValue>,
+    required: true,
+  },
+})
+
+const config = inject<TraceViewerConfig>(TRACE_VIEWER_CONFIG)!
+
+// TODO: This formatter is simply implemented for now
+const formattedValue = computed(() => {
+  const value = unwrapAnyValue(props.keyValue.value)
+
+  if (typeof value === 'string') {
+    return value
+  }
+
+  if (typeof value === 'boolean' || typeof value === 'number') {
+    return value.toString()
+  }
+
+  if (value instanceof Uint8Array) {
+    return JSON.stringify(Array.from(value))
+  }
+
+  return JSON.stringify(value)
+})
+
+// These are keys of attributes whose values are IDs of entities like services, routes, consumers, etc.
+const ENTITY_LINKED_ATTRIBUTE_KEYS = [
+  SPAN_ATTRIBUTES.KONG_SERVICE_ID.name,
+  SPAN_ATTRIBUTES.KONG_ROUTE_ID.name,
+  SPAN_ATTRIBUTES.KONG_CONSUMER_ID.name,
+  SPAN_ATTRIBUTES.KONG_PLUGIN_ID.name,
+  SPAN_ATTRIBUTES.KONG_UPSTREAM_ID.name,
+]
+
+const entityLink = ref<string | undefined>(undefined)
+const entityLinkData = shallowRef<EntityLinkData | undefined>(undefined)
+
+watchEffect(() => {
+  let entity: { entity: string; entityId: string } | undefined
```

Comment:
Should we declare `entity` and `entityLink` as computed values? IMO putting everything in a `watchEffect` makes the state dependency harder to understand.

Reply:
Agreed. I have renamed `entity` to `entityRequest` and made `entityLink` a ComputedRef for a better reading experience. I also refactored `watchEffect` into `watch` so that the dependencies will be more visible for lookup.